### ISSUE TITLE
GCP012 error is raised even though the service account is defined.

### ISSUE THREAD

**Issue by ruriky** [tfsec/tfsec#529]
**Describe the bug**
I have a TF configuration which disables the default node_pool in google_container_cluster and creates explicitly another pool using google_container_node_pool. TF creates a google service account and gives it name as a variable `var.service_account` to the node_pool config.

GCP012 fails even though I have defined service_account in google_container_node_pool resources node_config block.

The error disappears when I remove the variable and hard code the service_account value.

**Expected behavior**
After defining service_account in nodepool's node_config block as a variable, I should not see the error [GCP012]

**Screenshots**
In the error printout one can see that the service_account is defined.

```
Problem 1
  [GCP012][ERROR] Resource 'module.app-n1-standard-2-qa:google_container_node_pool.this' does not override the default service account. It is recommended to use a minimally privileged service account to run your GKE cluster.
  /builds/anders/terraform/terraform/modules/google/gke_nodepool/main.tf:30-64
      27 |     max_unavailable = 0
      28 |   }
      29 | 
      30 |   node_config {
      31 |     service_account = var.service_account
      32 | ...
```

**Comment by joonvena** [comment]
This will be fixed when this PR is merged:
https://github.com/tfsec/tfsec/pull/540

**Comment by gimartin-orng** [comment]
Same issue using google terraform module for gke
```
     376 |     service_account = lookup(
     377 |       each.value,
     378 |       "service_account",
     379 |       local.service_account,
     380 |     )
```

**Comment by owenrumney** [comment]
@gimartin-orng I think this is resolved in v0.38.2
